containers/podman-desktop
# WIP feat: Initial Welcome page

### What does this PR do?

Creates a new Welcome page that will be shown once for every user. As per the feature this is just a simple page for now, but could be expanded in future sprints to help the user install podman or do other first-time setup.

### Screenshot/screencast of this PR

<img width="884" alt="Screenshot 2023-03-02 at 4 13 13 PM" src="https://user-images.githubusercontent.com/19958075/222555033-ca9f58db-5a47-4135-93fa-fc05c82074ea.png">

### What issues does this PR fix or reference?

Issue #1334.

### How to test this PR?

The welcome should appear the first time you start and then never again. To see the welcome a second time, remove the welcome.version property from the configuration (e.g. .local/share/containers/podman-desktop/configuration/settings.json on Mac). 
(or if you just want to see the page again quickly in dev, add `<a href="/welcome">Welcome</a>` to another page)

Implementation notes:
- The new page is at a new route /welcome. The title bar is always shown and an if statement blocks the rest of the UI from appearing.
- I've added the podman desktop logo as images/DesktopIcon because I don't believe we've used it elsewhere, but will likely need it on other pages in the future.
- The welcome page has a blurred background image and a static list of supported container engines. It could still be tweaked but is very close to the proposed design.
- A welcome.version configuration property (not shown in settings) is used to store the fact that the welcome page has been shown. The value is set to 'initial' for now. My intention was that this could be set to the version of podman desktop or welcome page shown so in a future sprint we can decide when people who have seen the welcome should now see a newer/better version.
